### PR TITLE
Improve READMEs

### PR DESCRIPTION
**Pull request by godric**
# What?
 - Align main README content and format to match the backend README
 - Move instructions in a per-level README
 - Attempt to clarify the instructions following interview feedbacks
 - English

### CODE REVIEW

File: reliability/README.md [hunk 1]
```diff
@@ -1,103 +1,41 @@
...
-- Clean
-- Extensible
-- Reliable
+You can have a look at the higher levels, but please do the simplest thing that could work for the level you're currently solving.
 
-## Challenge
+The levels become more complex over time, so you will probably have to re-use some code and adapt it to the new requirements.
+A good way to solve this is by using OOP, adding new layers of abstraction when they become necessary and possibly write tests so you don't break what you have already done.
 
-The challenge needs to be resolved in Ruby.
```
Comment (marcgg) [inline reliability/README.md:20]: This is not required anymore?

File: reliability/README.md [hunk 2]
```diff
@@ -1,103 +1,41 @@
...
+Don't hesitate to write [shameless code](http://red-badger.com/blog/2014/08/20/i-spent-3-days-with-sandi-metz-heres-what-i-learned/) at first, and then refactor it in the next levels.
 
-Each level is a directory containing ruby executables and libraries that you'll have to use.
+For higher levels we are interested in seeing code that is clean, extensible and robust, so don't overlook edge cases, use exceptions where needed, ...
 
-You can't modify them.
+Please also note that:
 
-Your solution to each level needs to live in the level directory.
+- Running `$ ruby main.rb` from the level folder should generate the desired output, but of course feel free to add more files if needed.
```
Comment (marcgg) [inline reliability/README.md]: Is this correct?

File: reliability/level2/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Level 2
+
+## Summary
+
+The `level2` program will send log messages to a local HTTP server at http://localhost:3000/.
```
Comment (marcgg) [inline reliability/level2/README.md:5]: if there are multiple programs, then the initial brief of launching `$ ruby main` might be a bit unclear

File: reliability/level3/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Level 3
+
+## Summary
+
+The level3 program is the same as level2.
```
Comment (marcgg) [inline reliability/level3/README.md:5]: That doesn't feel right, there are some differences

File: reliability/README.md (hunk 1 above)
Comment (godric) [reply reliability/README.md:20]: Yes we would prefer Ruby, but if the candidate think he/she can produce something marginally better in another language we'll take it.
Finding Ruby developers is hard enough, we don't want to close any doors.

File: reliability/README.md (hunk 2 above)
Comment (godric) [reply reliability/README.md]: The implementation for each level should live in a new file, `main.rb` seems standard enough.

Maybe we should provide empty `main.rb` files in each folder ?

File: reliability/level3/README.md (hunk 4 above)
Comment (godric) [reply reliability/level3/README.md:5]: There isn't. The files are literally identical, but I think I understand where you're coming from, I'll add something in the base README.me about the level helpers.

File: reliability/level2/README.md (hunk 3 above)
Comment (godric) [reply reliability/level2/README.md:5]: Right, I should probably add something to make the distinction between the `levelX` (generator) and `main.rb` (solution)

File: reliability/level1/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Level 1
+
+## Summary
+
+The `level1` program it will write log messages into `./logs/#{id}.txt` files.
```
Comment (mickey) [inline reliability/level1/README.md]: ~it~

**Review by mickey — APPROVED** [review]
👍 I agree we could be clearer when stating than levelX is the same as levelY.

File: reliability/level4/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+The level4 program is the same as level2 and level3.
+
+## Instructions
+
+Write a simple HTTP server that serves the requests, parse and "enrich" the logs, and store them into a Redis `LIST` on a local Redis instance.
+
+The log "enrichment" is done using the provided `SlowComputation` module.
+
+**BEWARE** This module is pretty slow and the level4 program enforces a 100ms timeout.
```
Comment (tah91) [inline reliability/level4/README.md:13]: may be we can add something like "this means that any request lasting more than this timeout will fail, so we want an implementation that will avoid it."

Comment (tah91) [reply reliability/level4/README.md:13]: because it is not that clear, if you don't know the issue, in my opinion

**Pull request merged**